src/components/Sidebar/Sidebar.jsx

```diff
@@ -29,14 +29,20 @@ export const Sidebar = ({
     onLogoClick,
     isStickyHeader = true,
     isStickyFooter = true,
+    shouldCloseOnRouteChange = true,
 }) => {
     const [isLeftDrawerOpen, setIsLeftDrawerOpen] = useState(false);
     const [isRightDrawerOpen, setIsRightDrawerOpen] = useState(false);
 
+    // eslint-disable-next-line no-undef
+    const location = typeof window === "undefined" ? undefined : window.location;
+
     useEffect(() => {
-        setIsLeftDrawerOpen(false); // Close the drawer if notifications changes
-        setIsRightDrawerOpen(false);
-    }, [notifications]);
+        if (shouldCloseOnRouteChange && location?.search?.includes("redirect=true")) {
+            setIsLeftDrawerOpen(false);
+            setIsRightDrawerOpen(false);
+        }
+    }, [location.search, shouldCloseOnRouteChange]);
```

Comment:
I think its better to close the drawer from x2-seller app. since this ui-kit component might be used across different apps. can you try to trigger the close event, when we click on decline / accept purchase

Also, to prevent closing this drawer on notifications changes, we can pass additional props `shouldCloseOnChanges` - pass this as false from x2-seller app. 

@rushi can you also review this once

Reply:
done, could you please take another look
also, since we are preventing closing on notification changes, we need to manually close it after opening the purchase,  to handle the case which was fixed by adding close-on-change logic previously (when we open a purchase from the purchases page), this logic added as part of the x2-seller PR https://github.com/xola/x2-seller/pull/3773